### PR TITLE
Solved Edit breaks by broken Short Names

### PR DESCRIPTION
**Pull request by pranayjoshi**
Solved #5407 . Everytime a User enters an ineligible character, the code will restrict from entering the next character. This will continue until the user removes the last character.

### CODE REVIEW

**Comment by stevepiercy** [comment]
@pranayjoshi tests should pass on your development environment before opening a pull request. See https://6.docs.plone.org/volto/contributing/index.html#code-quality

**Comment by pranayjoshi** [comment]
@stevepiercy I updated my commits. Lemme know if it requires further changes.

File: news/5407.documentation
Comment (stevepiercy) [inline news/5407.documentation:1]: This file should be renamed `5407.bugfix` because it fixes a bug. See https://6.docs.plone.org/contributing/index.html#change-log-entry

Also we like descriptive descriptions in our change logs. "Fixed or Solved issue title" is useless because it does not describe the bug and its fixed state. See item 2 in https://6.docs.plone.org/contributing/first-time.html#create-a-pull-request-from-your-fork.

```suggestion
When you add a page, only lowercase letters (a-z) without accents, numbers (0-9), and the characters "-", "_", and "." are enforced for the short name. Also after creating the page, you can now edit it. Previously you could enter invalid characters and create the page, and then you could not edit it. @pranayjoshi
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Thank you!

I just review docs and comments. A core contributor should review this PR and provide further feedback.

**Comment by pranayjoshi** [comment]
Changes made.

**Review by stevepiercy — APPROVED** [review]

**Comment by stevepiercy** [comment]
Superseded by https://github.com/plone/volto/pull/5443

**Pull request closed without merging**
